src/node/building_blocks.ml

```diff
@@ -102,18 +108,38 @@ let should_start_new_epoch last_state_root_update current_time =
     \    The block producer uses this function to determine when to send a\n\
     \    block with an updated state root hash.\n"]
 
+(** Can only included a tezos operation if enough time has already elapsed *)
+let can_included_tezos_operation ~current_time pending_operation =
+  current_time -. pending_operation.Node.requested_at
+  > minimum_waiting_period_for_tezos_operation
+
 let produce_block state =
+  let current_time = Unix.time () in
   let start_new_epoch =
     should_start_new_epoch state.Node.protocol.last_state_root_update
-      (Unix.time ()) in
+      current_time in
   let next_state_root_hash =
     if start_new_epoch then
       let%some snapshot = Snapshots.get_next_snapshot state.snapshots in
       Some snapshot.hash
     else
       None in
+  let operations =
+    List.filter_map
+      (fun pending_operation ->
+        let operation = pending_operation.Node.operation in
+        match operation with
+        | Operation.Core_tezos _ ->
+          if can_included_tezos_operation ~current_time pending_operation then
+            Some operation
+          else
+            None
+        | Core_user _
+        | Consensus _ ->
+          Some operation)
```

Comment:
I mean, not exactly a proposal, but a question. Why not like this?

```ocaml
match operation with
| Consensus _
| Operation.Core_tezos _ when can_included_tezos_operation ~current_time pending_operation ->
   Some operation
| _ -> None
 ```

Reply:
The difference is that in one you can read the match separated from the conditional.

In general I avoid this pattern when I'm talking about a single case, the way that I read the current code is, `if it is a Core_tezos then do this thing` while the one proposed by you is, `if it's a Core_tezos and can be included then do this thing`. 

Also in general avoid catch all `_`.